backends/events/sdl/sdl-events.cpp

```diff
@@ -1077,3 +1091,434 @@ uint32 SdlEventSource::obtainUnicode(const SDL_Keysym keySym) {
+                int32_t tdDeltaY = posY - _touchFingers[port][iii].tdPosY;
+                uint32_t sqTdDistance = (tdDeltaX * tdDeltaX) + (tdDeltaY * tdDeltaY);
+
+                if (ConfMan.getBool("touchpad_mouse_mode")) {
+                    // Touchpad mode -> Use current mouse position
+                    posX = _mouseX;
+                    posY = _mouseY;
+                }
+
+                if (timeDelta <= static_cast<int32_t>(_touchMaxTapTime) && 
+                    sqTdDistance <= _touchMaxTapSqDist) {
+                    // Short tap -> Mouse click
+                    switch (touchCount) {
+                        case 1:
+                        { // One touch point -> Left Click
+                            event->type = SDL_MOUSEBUTTONDOWN;
+                            event->button.button = SDL_BUTTON_LEFT;
+                            event->button.x = posX;
+                            event->button.y = posY;
+                            _touchClickTime[port][TOUCH_BUTTON_LEFT] = event->tfinger.timestamp;
+                        } break;
+                        case 2:
+                        { // Two touch points -> Right Click
+                            event->type = SDL_MOUSEBUTTONDOWN;
+                            event->button.button = SDL_BUTTON_RIGHT;
+                            event->button.x = posX;
+                            event->button.y = posY;
+                            _touchClickTime[port][TOUCH_BUTTON_RIGHT] = event->tfinger.timestamp;
+                        } break;
+                        // TODO - Can add more gestures here...
+                        default:
+                            break;
+                    }
+                }
+            } break;
+            case TOUCH_DRAG_TWO: 
+            {
+                // Two-finger drag
+                if (touchCount == 1) {
+                    // Ending drag
+                    
+                    if (ConfMan.getBool("touchpad_mouse_mode")) {
+                        // Touchpad mode -> Use current mouse position
+                        posX = _mouseX;
+                        posY = _mouseY;
+                    }
+
+                    event->type = SDL_MOUSEBUTTONUP;
+                    event->button.button = SDL_BUTTON_LEFT;
+                    event->button.x = posX;
+                    event->button.y = posY;
+                    _touchDragType[port] = TOUCH_DRAG_NONE;
+                }
+            } break;
+            case TOUCH_DRAG_THREE: 
+            {
+                // Three-finger drag
+                if (touchCount == 1) {
+                    // Ending drag
+                    
+                    if (ConfMan.getBool("touchpad_mouse_mode")) {
+                        // Touchpad mode -> Use current mouse position
+                        posX = _mouseX;
+                        posY = _mouseY;
+                    }
+
+                    event->type = SDL_MOUSEBUTTONUP;
```

Comment:
I am thinking, does it make sense to hardcode these to mouse events? Maybe it would be better to expose them outside and then map to the mouse as default key mapping?